astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -974,52 +981,56 @@ def test_data_out_of_range(parallel, reader):
     """
     # Python reader and strtod() are expected to return precise results
     rtol = 1.e-30
-    if reader > 1:
-        rtol = 1.e-15
-    # passing fast_reader dict with parametrize does not work!
-    if reader > 0:
-        fast_reader = {'parallel': parallel, 'use_fast_converter': reader > 1}
-    else:
-        fast_reader = False
+
+    # update fast_reader dict
+    if fast_reader:
+        fast_reader['parallel'] = parallel
+        if fast_reader.get('use_fast_converter'):
+            rtol = 1.e-15
+
     if parallel:
-        if reader < 1:
+        if not fast_reader:
             pytest.skip("Multiprocessing only available in fast reader")
         elif TRAVIS:
             pytest.xfail("Multiprocessing can sometimes fail on Travis CI")
 
-    fields = ['10.1E+199', '3.14e+313', '2048e+306', '0.6E-325', '-2.e345']
-    values = np.array([1.01e200, np.inf, np.inf, 0.0, -np.inf])
-    t = ascii.read(StringIO(' '.join(fields)), format='no_header', guess=False,
-                   fast_reader=fast_reader)
+    fields = [ '10.1E+199', '3.14e+313', '2048e+306', '0.6E-325', '-2.e345' ]
+    values = np.array([ 1.01e200, np.inf, np.inf, 0.0, -np.inf ])
+    t = ascii.read(StringIO(' '.join(fields)), format='no_header',
+                   guess=guess, fast_reader=fast_reader)
     read_values = np.array([col[0] for col in t.itercols()])
     assert_almost_equal(read_values, values, rtol=rtol, atol=1.e-324)
 
     # test some additional corner cases
-    fields = ['.0101E202', '0.000000314E+314', '1777E+305', '-1799E+305', '0.2e-323',
-               '2500e-327', ' 0.0000000000000000000001024E+330']
-    values = np.array([1.01e200, 3.14e307, 1.777e308, -np.inf, 0.0, 4.94e-324, 1.024e308])
-    t = ascii.read(StringIO(' '.join(fields)), format='no_header', guess=False,
-                   fast_reader=fast_reader)
+    fields = [ '.0101E202', '0.000000314E+314', '1777E+305', '-1799E+305', '0.2e-323',
+               '5200e-327', ' 0.0000000000000000000001024E+330' ]
```

Comment:
Why was this changed from 2500e-327 to 5200e-327?

Reply:
Hmm, not sure about the reason - the `float64` for both is ca. `4.94e-324` - or is it possible the former could have parsed as 0 on some architectures? Will probably try both...